### PR TITLE
feat(loader): support multiple versions of the same dependency

### LINKED ISSUE #215: [Feature] support multiple versions of the same dependency
This should be possible by setting up the [luarocks loader](https://github.com/luarocks/luarocks/wiki/Using-LuaRocks#multiple-versions-using-the-luarocks-package-loader).

We may need to provide a `LUAROCKS_CONFIG` file and a way to get the path to the loader in the luarocks installation.

### PR DESCRIPTION
**Pull request by mrcjkb**
Closes #215

### CODE REVIEW

File: plugin/rocks.lua [hunk 1]
```diff
@@ -6,6 +6,25 @@ local nio = require("nio")
 local adapter = require("rocks.adapter")
 local config = require("rocks.config.internal")
 
+-- Initialize the luarocks loader
+if config.enable_luarocks_loader then
+    local sc = vim.system({ config.luarocks_binary, "which", "luarocks.loader" }):wait()
+    local luarocks_path = sc.stdout and sc.stdout:match("(%S+)loader.lua")
```
Comment (vhyrro) [inline plugin/rocks.lua:12]: I assume we are doing the whole "luarocks which" check as we also permit users to use their system installation of luarocks, right?

File: plugin/rocks.lua [hunk 2]
```diff
@@ -6,6 +6,25 @@ local nio = require("nio")
...
+    if luarocks_path then
+        package.path = package.path .. ";" .. luarocks_path .. "?.lua"
+        vim.uv.os_setenv("LUAROCKS_CONFIG", config.luarocks_config)
+        local ok, err = pcall(function()
+            require("luarocks.loader")
+        end)
```
Comment (vhyrro) [inline plugin/rocks.lua:18]:
```suggestion
        local ok, err = pcall(require, "luarocks.loader")
```

This bit can be shortened :)

File: plugin/rocks.lua [hunk 3]
```diff
@@ -6,6 +6,25 @@ local nio = require("nio")
 local adapter = require("rocks.adapter")
 local config = require("rocks.config.internal")
 
+-- Initialize the luarocks loader
+if config.enable_luarocks_loader then
+    local sc = vim.system({ config.luarocks_binary, "which", "luarocks.loader" }):wait()
+    local luarocks_path = sc.stdout and sc.stdout:match("(%S+)loader.lua")
+    if luarocks_path then
+        package.path = package.path .. ";" .. luarocks_path .. "?.lua"
+        vim.uv.os_setenv("LUAROCKS_CONFIG", config.luarocks_config)
```
Comment (vhyrro) [inline plugin/rocks.lua:15]:
```suggestion
        vim.env.LUAROCKS_CONFIG = config.luarocks_config
```

nit: would this not be the simpler approach, or does it have some drawbacks I'm not aware of? :)

File: plugin/rocks.lua (hunk 1 above)
Comment (mrcjkb) [reply plugin/rocks.lua]: Yes.
I've just added a change that doesn't call `vim.system` if the luarocks path `<rocks_config>/bin/luarocks`.

**Pull request merged**